frontend/src/bundles/home/components/main-content/main-content.tsx

```diff
@@ -1,12 +1,37 @@
-import { Box } from '~/bundles/common/components/components.js';
+import {
+    Box,
+    Loader,
+    Overlay,
+} from '~/bundles/common/components/components.js';
+import { DataStatus } from '~/bundles/common/enums/enums.js';
+import {
+    useAppDispatch,
+    useAppSelector,
+    useEffect,
+} from '~/bundles/common/hooks/hooks.js';
+import { actions as homeActions } from '~/bundles/home/store/home.js';
 
 import { VideoSection } from '../components.js';
 
 const MainContent: React.FC = () => {
+    const dispatch = useAppDispatch();
+
+    const { videos, dataStatus } = useAppSelector(({ home }) => home);
```

Comment:
It would be better to return only the fields you need instead of returning the entire home store.